routes/user.js

```diff
@@ -0,0 +1,166 @@
+      !req.query.departureDate ||
+      !req.query.roomId ||
+      !req.params.userId
+    )
+      return res.status(400).send(`Invalid Request`);
+
+    const checkIfBookingExicts = await Booking.findOne({
+      arrivalDate: { $gte: req.query.arrivalDate },
+      departureDate: { $lt: req.query.departureDate },
+      roomId: req.params.userId,
+    });
+
+    if (checkIfBookingExicts && checkIfBookingExicts.isApproved)
+      return res.status(400).send(`The room is not avaible on those dates !`);
+
+    const booking = new Booking({
+      userId: req.params.userId,
+      roomId: req.query.roomId,
+      arrivalDate: req.query.arrivalDate,
+      departureDate: req.query.departureDate,
+    });
+
+    const newBooking = await booking.save();
+    res.status(201).send(newBooking);
+  }
+);
+
+router.delete(
+  "/requestToBook/:bookingId",
+  [validateID, validateParamsIDwithToken(["bookingId"]), authz(["Client"])],
+  async (req, res) => {
+    //requestToBook/:bookingId
+    if (!req.params.bookingId) return res.status(400).send(`Invalid Request`);
+
+    const checkIfBookingExicts = await Booking.findOne({
+      _id: req.params.bookingId,
+    });
+
+    if (!checkIfBookingExicts)
+      return res
+        .status(400)
+        .send(
+          `The booking with the id: ${req.params.bookingId} does not exists  !`
+        );
+
+    const bookingDelete = await checkIfBookingExicts.deleteOne({
+      _id: checkIfBookingExicts._id,
+    });
+    res.status(200).send(bookingDelete);
+  }
+);
+
+router.get(
+  "/requestToBook/:bookingId",
+  [validateID, validateParamsIDwithToken(["bookingId"]), authz(["Client"])],
+  async (req, res) => {
+    //requestToBook/:bookingId
+    if (!req.params.bookingId) return res.status(400).send(`Invalid Request`);
+
+    const booking = await Booking.findOne({
+      _id: req.params.bookingId,
+    })
+      .populate({ path: "userId", select: "firstName, lastName, email" })
+      .populate({ path: "roomId", select: "type, number, description" });
+
+    if (!booking)
+      return res
+        .status(400)
+        .send(
+          `The booking with the id: ${req.params.bookingId} does not exists  !`
+        );
+    booking;
+
+    res.status(200).send(booking);
+  }
+);
+
+router.put(
+  "/requestToBook/:bookingId",
+  [validateID, validateParamsIDwithToken(["bookingId"]), authz(["Client"])],
+  async (req, res) => {
+    //requestToBook/:id?arrivalDate=YYYY-MM-DD&&departureDate=YYYY-MM-DD&&roomId=:id
+    if (
+      !req.query.arrivalDate ||
+      !req.query.departureDate ||
+      !req.query.roomId ||
+      !req.params.bookingId
+    )
+      return res.status(400).send(`Invalid Request !`);
+
+    const booking = await Booking.findOne({ _id: req.params.bookingId });
+
+    if (!booking)
+      return res
+        .status(400)
+        .send(
+          `The booking with the id: ${req.params.bookingId} does not exists  !`
+        );
+
+    const RoomAvaible = await Booking.findOne({
```

Comment:
Please use camel case with variables, also use english properly. The variable naming is `availableRoom`.